prow/crier/reporters/gerrit/reporter.go

```diff
@@ -353,7 +406,25 @@ func (c *Client) Report(ctx context.Context, logger *logrus.Entry, pj *v1.ProwJo
 	}
 
 	logger.Infof("Review Complete, reported jobs: %s", jobNames(toReportJobs))
-	return toReportJobs, nil, nil
+
+	// If return here, the shardedLock will be released, and other threads that
+	// are from the same PR will still not understand that it's already
+	// reported, as the change of previous report state happens only after the
+	// returning of current function from the caller.
+	// Ideally the previous report state should be changed here.
+	logger.WithField("job-count", len(toReportJobs)).Info("Reported job(s), now will update pj(s).")
+	var err error
+	for _, pjob := range toReportJobs {
+		// if pjob.Status.PrevReportStates == nil {
+		// 	pj.Status.PrevReportStates = map[string]v1.ProwJobState{}
+		// }
+		if err = criercommonlib.UpdateReportStateWithRetries(ctx, pjob, logger, c.pjclientset, c.GetName()); err != nil {
+			logger.WithError(err).Error("Failed to update report state on prowjob")
+		}
+	}
+
+	// Let caller know that we are done with this job.
+	return []*v1.ProwJob{pj}, nil, err
```

Comment:
`pj` needs to be reported before the lock is released and should already be included in `toReportJobs` so I don't think we should be returning it here.

Reply:
Agreed. I think it should be fine not to return anything here